src/compiler/parser.ts

```diff
@@ -8365,21 +8379,29 @@ namespace Parser {
             identifier = parseIdentifier();
         }
 
-        let isTypeOnly = false;
+        let phaseModifier: ImportPhaseModifier | undefined;
         if (
             identifier?.escapedText === "type" &&
             (token() !== SyntaxKind.FromKeyword || isIdentifier() && lookAhead(nextTokenIsFromKeywordOrEqualsToken)) &&
             (isIdentifier() || tokenAfterImportDefinitelyProducesImportDeclaration())
         ) {
-            isTypeOnly = true;
+            phaseModifier = SyntaxKind.TypeKeyword;
             identifier = isIdentifier() ? parseIdentifier() : undefined;
         }
+        else if (identifier?.escapedText === "defer" && token() !== SyntaxKind.FromKeyword) {
+            phaseModifier = SyntaxKind.DeferKeyword;
+            identifier = undefined;
+            if (isIdentifier()) {
+                parseErrorAtCurrentToken(Diagnostics.Default_imports_aren_t_allowed_for_deferred_imports);

```

Comment:
I think it would make more sense to treat this as a grammar error in the checker